view/src/main/java/cmm/apps/esmorga/view/eventdetails/EventDetailsView.kt

```diff
@@ -0,0 +1,154 @@
+package cmm.apps.esmorga.view.eventdetails
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.aspectRatio
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.automirrored.filled.ArrowBack
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import cmm.apps.designsystem.EsmorgaButton
+import cmm.apps.esmorga.view.R
+import cmm.apps.esmorga.view.eventdetails.model.EventDetailsEffect
+import cmm.apps.esmorga.view.eventdetails.model.EventDetailsUiState
+import cmm.apps.esmorga.view.navigation.openNavigationApp
+import cmm.apps.esmorga.view.theme.EsmorgaTheme
+import cmm.apps.esmorga.view.theme.Sepia
+import coil.compose.AsyncImage
+import coil.request.ImageRequest
+import org.koin.androidx.compose.koinViewModel
+
+
+@Composable
+fun EventDetailsScreen(eventId: String, onBackPressed: () -> Unit, edvm: EventDetailsViewModel = koinViewModel()) {
+    val uiState: EventDetailsUiState by edvm.uiState.collectAsStateWithLifecycle()
+    edvm.loadEventDetails(eventId)
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/cmm-apps-android/EsmorgaAndroid/pull/5#issuecomment-2194728801>#267d9d</a><span> - 06/27/2024, 01:47 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The 'loadEventDetails' function is called directly within the 'EventDetailsScreen' composable function. This can lead to multiple invocations of 'loadEventDetails' if the composable recomposes, which can cause performance issues and unintended side effects.
</div>
<div id="fix">
<b>Fix:</b> Move the 'loadEventDetails' call to a 'LaunchedEffect' block to ensure it is only called once when the composable is first composed.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -41,7 +41,7 @@
      val uiState: EventDetailsUiState by edvm.uiState.collectAsStateWithLifecycle()
 -    edvm.loadEventDetails(eventId)
      val context = LocalContext.current
      LaunchedEffect(Unit) {
 +        edvm.loadEventDetails(eventId)
          edvm.effect.collect { eff ->
              when (eff) {
                  is EventDetailsEffect.NavigateToLocation -> {
```
</div>
</details>
</div>
<br>


</div>

Reply:
Is there any way to make the ViewModel do this?

Reply:
https://media1.tenor.com/m/lTHbW-i6vtIAAAAC/magic-shia-la-beouf.gif